packages/admin/dashboard/src/routes/orders/order-detail/components/order-summary-section/shipping-info-popover.tsx

```diff
@@ -0,0 +1,46 @@
+import { InformationCircleSolid } from "@medusajs/icons"
+import { AdminOrderShippingMethod } from "@medusajs/types"
+import { Badge, Tooltip } from "@medusajs/ui"
+
+type ShippingInfoPopoverProps = {
+  shippingMethod: AdminOrderShippingMethod
+}
+
+function ShippingInfoPopover({ shippingMethod }: ShippingInfoPopoverProps) {
+  const shippingDetail = shippingMethod?.detail
+
+  if (!shippingDetail) {
+    return
+  }
+
+  let rmaType = "Return"
+  let rmaId = shippingDetail.return_id
+
+  if (shippingDetail.claim_id) {
+    rmaType = "Claim"
+    rmaId = shippingDetail.claim_id
+  }
+
+  if (shippingDetail.exchange_id) {
+    rmaType = "Exchange"
+    rmaId = shippingDetail.exchange_id
+  }
+
+  if (typeof shippingDetail !== "object" || !rmaId) {
```

Comment:
q: why do we check `shippingDetail` here and on L:12, it should be an object at this point otherwise some of the lines above would throw